.github/actions/check-artifact-hash/action.yml

```diff
@@ -0,0 +1,56 @@
+name: Check the Git hash in an artifact if it exists
+
+inputs:
+  package_path:
+    description: 'The path to the package for which the artifact should be checked'
+    required: true
+    type: string
+
+outputs:
+  artifact_name:
+    description: "The name of the generated artifact"
+    value: ${{ steps.parse.outputs.artifact_name }}
+  needs_rebuild:
+    description: "Does the package need to be rebuilt?"
+    value: ${{ steps.compare.outputs.rebuild }}
+  hash:
+    description: "The hash to be saved as artifact"
+    value: ${{ steps.compare.outputs.hash }}
+
+runs:
+  using: composite
+  steps:
+    - id: parse
+      name: Parse artifact name
+      run: |
+        COMP_PATH=${{ inputs.package_path }}
+        COMP_PATH=${COMP_PATH//\//_}
+        echo "artifact_name=${COMP_PATH:2}" >> $GITHUB_OUTPUT
+      shell: bash
+
+    - uses: dawidd6/action-download-artifact@v2
+      with:
+        workflow: build-release.yaml
+        workflow_conclusion: success
+        branch: ${{ github.ref_name }}
```

Comment:
Why restrict to the branch? Without we could use this to build on feature branches with minimal rebuild

Reply:
You mean that if we open a PR on a feature branch we could check the diff to the target branch? not a bad idea :+1: Should I make this branch an input to the action?

Reply:
Basically, but I think just using the target branch opens other issue. Really we should have a way to track multiple hashes 😕

Reply:
Given that for now HC and dynamic-components only run that workflow on develop and main, this is good to go for now no? I see the advantages of multiple hashes for the future though